### DISCUSSION TITLE
Warning Icon for non-mandatory PR checks instead of 'X'

### DISCUSSION THREAD

**Discussion by ed0906** [community/community#35424] [Actions]
In our team we use Github Actions to run Checks on PRs. 
Some of the checks block a merge when they fail. Some other checks can fail but don't block a merge. 
Github already offers this functionality which is great. 

The trouble we have is the checks that fail which don't block merges still have a big ❌ beside them which can look like there's a problem, even when the check could just be prompting the **reviewer** to take action rather than the developer e.g. Review a particular file in more detail for backwards compatibility. 
It also results in that PR having an ❌ against it on summary screens such as `https://github.com/{org}/{repo}/pulls`.
The above all acts as a deterrent to potential reviewers as it may look like the PR isnt ready for review.

I would like to propose the introduction of an option for PR checks to fail with a user configurable ❌ or an ⚠️ icon

----

An example of our use case
![image](https://user-images.githubusercontent.com/4864094/194539564-eb36c827-d96c-4046-ad82-f6fab2485fa8.png)

We have a Check which verifies if an open api spec has changed as part of a PR. 
If that file has changed, the check currently fails which should prompt a reviewer to look at the changes to that file in detail for backwards compatibility.
The trouble is, from experience, when people see a ❌  they often think "The PR isn't ready for review" and stop their review, or hold off testing it.

If instead these optional checks could be configured to fail with ⚠️ , and that icon propagates to summary screens (providing all other mandatory checks pass) then I believe reviewers would be more likely to review PRs which are in fact ready for review, but have warnings/actions attached.

**Comment by knocte** [comment]
> Some other checks can fail but don't block a merge.

How did you achieve this?

**Reply by ed0906** [reply]
I thought I had this set up in branch protection settings, but it could be that because I'm a project admin that I can merge anyway...

If thats not a feature already it probably should be as well

**Reply by ed0906** [reply]
Ignore the above this is possible.

Mandatory checks that block a merge: Use branch protection settings for master/whatever your main branch is called
Optional checks that dont block a merge: all other pr checks that arent in branch protection settings


The trouble is that even though what I call the Optional checks dont block a merge, they still have a big scary red ❌ next to them so whoever comes to review/merge may feel like something needs rectifying, when in fact it could just be a call to action e.g. "review the api spec", or "tell another team something is going to change"

**Comment by rousseldenis** [comment]
I should say +1 for such feature

related to https://github.com/isaacs/github/issues/1465

**Reply by vpaulson** [reply]
Any chance this will ever get looked at? In particular the scary red X on the main `/pulls` page of the Repo confuses people thinking those pull requests are not mergable.

**Comment by fsmeier** [comment]
+1 - I thought we had this already, but maybe it was gitlab....

**Comment by zsolt-p** [comment]
+1 this would be great

**Comment by lizziemac** [comment]
Adding to this, my team has an action that gates merging a PR if not all relevant code owners have approved. As a reviewer, even when using the `action required` feature, and everything the developer is responsible for passes, the PR itself has the red ❌. This means that reviewers, who we need to review it to unblock the dev, will ignore the PR, or check every failing PR because they don't know which just needs a review, and which is actually failing. I agree that a ⚠️ would be great for this and the above use cases.


<img width="734" height="156" alt="Screenshot 2026-08-13 at 10 32 57" src="https://github.com/user-attachments/assets/1378a8ca-ce47-4407-9248-a605002ce5c0" />
